### PR TITLE
[ENG-6043] OwncloudAPI Storage Implementation

### PR DESCRIPTION
**Pull request by sh-andriy**
Implemented **`Owncloud Storage Addon`**. For more details look at ticket **`ENG-6043`**

### CODE REVIEW

File: addon_service/external_service/storage/models.py [hunk 1]
```diff
@@ -40,6 +40,8 @@ class ExternalStorageService(ExternalService):
     @property
     def supported_features(self) -> list[SupportedFeatures]:
         """get the enum representation of int_supported_features"""
+        if self.int_supported_features is None:
+            return SupportedFeatures(0)
```
Comment (opaduchak) [inline addon_service/external_service/storage/models.py:44]: can you elaborate on this change please?

File: addon_imps/storage/owncloud.py [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+            if item_type is not None and _item_type != item_type:
+                continue
+
+            displayname_element = response_element.find(".//d:displayname", ns)
+            if displayname_element is not None and displayname_element.text:
+                displayname = displayname_element.text
+            else:
+                displayname = path.rstrip("/").split("/")[-1]
+
```
Comment (opaduchak) [inline addon_imps/storage/owncloud.py:190]: it seems that these lines are nearly identical to 134-146, if it isn't much trouble, please make it more DRY

File: addon_imps/storage/owncloud.py [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+            <d:prop>
+                <d:displayname/>
+            </d:prop>
+        </d:propfind>"""
+
+    def _build_propfind_allprops(self) -> str:
+        return """<?xml version="1.0" encoding="UTF-8"?>
+        <d:propfind xmlns:d="DAV:">
+            <d:allprop/>
+        </d:propfind>"""
```
Comment (opaduchak) [inline addon_imps/storage/owncloud.py:106]: Maybe extract this into Enum or at least keep them separate from Imp as they impose some clutter

**Review by opaduchak — CHANGES_REQUESTED** [review]

File: addon_imps/storage/owncloud.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+            raise ValueError("current-user-principal not found in response")
+
+    def _parse_displayname(self, response_xml: str) -> str:
+        ns = {"d": "DAV:"}
+        root = ET.fromstring(response_xml)
+        displayname = root.find(".//d:displayname", ns)
+        if displayname is not None and displayname.text:
+            return displayname.text
+        else:
+            raise ValueError("displayname not found in response")
```
Comment (opaduchak) [inline addon_imps/storage/owncloud.py:168]: duplication of `_parse_current_user_principal`

**Review by opaduchak — CHANGES_REQUESTED** [review]

**Review by brianjgeiger — APPROVED** [review]
XML interface? What a pain!

**Pull request merged**